### ISSUE TITLE
Headless GUI Test Fails on Windows

### ISSUE THREAD

**Issue by AdamCarroll** [VocabHunter/VocabHunter#39]
The headless GUI test that works on Mac and Linux, fails on Windows.  If instead the build is launched with the `-PnoHeadless`, the build works without problems.

This was tested with OpenJDK 11.0.2 (AdoptOpenJDK) on Windows 10.  The error is as follows:
~~~
C:\Users\vagrant\Desktop\GitRepos\VocabHunter>gradlew :gui:test --tests io.github.vocabhunter.gui.main.GuiTest

> Task :gui:test
#
# A fatal error has been detected by the Java Runtime Environment:
#
#  EXCEPTION_ACCESS_VIOLATION (0xc0000005) at pc=0x00007ffa34584879, pid=2644, tid=3040
#
# JRE version: OpenJDK Runtime Environment (11.0.2+9) (build 11.0.2+9)
# Java VM: OpenJDK 64-Bit Server VM (11.0.2+9, mixed mode, tiered, compressed oops, serial gc, windows-amd64)
# Problematic frame:
# C  [javafx_font.dll+0x4879]
#
# No core dump will be written. Minidumps are not enabled by default on client versions of Windows
#
# An error report file with more information is saved as:
# C:\Users\vagrant\Desktop\GitRepos\VocabHunter\gui\hs_err_pid2644.log
#
# If you would like to submit a bug report, please visit:
#   https://github.com/AdoptOpenJDK/openjdk-build/issues
# The crash happened outside the Java Virtual Machine in native code.
# See problematic frame for where to report the bug.
#

Unexpected exception thrown.
org.gradle.internal.remote.internal.MessageIOException: Could not write '/127.0.0.1:50886'.
        at org.gradle.internal.remote.internal.inet.SocketConnection.flush(SocketConnection.java:135)
        at org.gradle.internal.remote.internal.hub.MessageHub$ConnectionDispatch.run(MessageHub.java:325)
        at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:63)
        at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:46)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
        at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:55)
        at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: java.io.IOException: An existing connection was forcibly closed by the remote host
        at java.base/sun.nio.ch.SocketDispatcher.write0(Native Method)
        at java.base/sun.nio.ch.SocketDispatcher.write(SocketDispatcher.java:51)
        at java.base/sun.nio.ch.IOUtil.writeFromNativeBuffer(IOUtil.java:113)
        at java.base/sun.nio.ch.IOUtil.write(IOUtil.java:58)
        at java.base/sun.nio.ch.IOUtil.write(IOUtil.java:50)
        at java.base/sun.nio.ch.SocketChannelImpl.write(SocketChannelImpl.java:466)
        at org.gradle.internal.remote.internal.inet.SocketConnection$SocketOutputStream.writeWithNonBlockingRetry(SocketConnection.java:273)
        at org.gradle.internal.remote.internal.inet.SocketConnection$SocketOutputStream.writeBufferToChannel(SocketConnection.java:261)
        at org.gradle.internal.remote.internal.inet.SocketConnection$SocketOutputStream.flush(SocketConnection.java:255)
        at org.gradle.internal.remote.internal.inet.SocketConnection.flush(SocketConnection.java:133)
        ... 7 more

> Task :gui:test FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':gui:test'.
> Process 'Gradle Test Executor 2' finished with non-zero exit value 1
  This problem might be caused by incorrect test process configuration.
  Please refer to the test execution section in the User Manual at https://docs.gradle.org/5.2.1/userguide/java_testing.html#sec:test_execution

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

Deprecated Gradle features were used in this build, making it incompatible with Gradle 6.0.
Use '--warning-mode all' to show the individual deprecation warnings.
See https://docs.gradle.org/5.2.1/userguide/command_line_interface.html#sec:command_line_warnings

BUILD FAILED in 31s
10 actionable tasks: 1 executed, 9 up-to-date
~~~

**Comment by AdamCarroll** [comment]
The same problem has been seen with the Santulator build on Windows: https://github.com/Santulator/Santulator/issues/14

**Comment by AdamCarroll** [comment]
This problem has been fixed now by updating the versions of JavaFX and Monocle in use in VocabHunter.  You can see the change here: 3a0e9e6c22977fc02a30498f00bc13c5fa8ded2c

It was also necessary to work around a character set problem to get the test working on Windows.  The fix is here: 6d3c6159d4ee3ef71992fef22e54d64f799244f7
